### PR TITLE
Modernize step-52

### PR DESCRIPTION
**Pull request by masterleinad**
Part of #6774
- use ranged-based for loops
- avoid `std::bind`
- make some member variables `const`

### CODE REVIEW

File: examples/step-52/step-52.cc [hunk 1]
```diff
@@ -354,11 +348,10 @@ namespace Step52
   //   - compute $z=\left(M-\tau \frac{\partial f}{\partial y}\right)^{-1} tmp =
   //   \left(M-\tau \frac{\partial f}{\partial y}\right)^{-1} My$
   //   - return z.
-  Vector<double> Diffusion::id_minus_tau_J_inverse(const double          time,
+  Vector<double> Diffusion::id_minus_tau_J_inverse(const double /*time*/,
```
Comment (bangerth) [inline examples/step-52/step-52.cc]: can you manually align the commented out variable to the level of the ones below? Or does clang-format undo it?

File: examples/step-52/step-52.cc [hunk 2]
```diff
@@ -503,10 +496,9 @@ namespace Step52
     for (unsigned int i = 0; i < n_time_steps; ++i)
       {
         time = explicit_runge_kutta.evolve_one_time_step(
-          std::bind(&Diffusion::evaluate_diffusion,
-                    this,
-                    std::placeholders::_1,
-                    std::placeholders::_2),
+          [this](const double time, const Vector<double> &y) {
+            return this->evaluate_diffusion(time, y);
+          },
```
Comment (bangerth) [inline examples/step-52/step-52.cc]: neat!

File: examples/step-52/step-52.cc (hunk 1 above)
Comment (masterleinad) [reply examples/step-52/step-52.cc]: No, I can't manually align the commented variable.

**Pull request merged**